api4/post.go

```diff
@@ -185,6 +194,17 @@ func getPostsForChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		mlog.Error("Failed to prepare posts for getPostsForChannel response", mlog.Any("err", err))
 	}
 
+	if nextPostId == "" {
+		nextPostId = c.App.GetNextPostFromPostList(clientPostList)
```

Comment:
@crspeller How much impact do you think an extra DB call or two will have here? This way of doing it is the most clear, but it may not be the most performance

I have an idea for how we can remove the extra DB call for at least the get posts before/after API calls although it's a bit tricky, and it'll require a few other changes in the app layer. Basically, we get one more post than we need, remove it from the list, and use that as the next/previous post ID. I was planning on doing something similar on the client-side to fake this behaviour on older servers.

We could also create a ticket to revisit this later since we're stopping work on the scrolling overhaul for a few weeks